File: leetcode/2011/final-value-of-variable-after-performing-operations.cpp
#include "string"
#include "vector"

class Solution {
 public:
  int finalValueAfterOperations(std::vector<std::string>& operations) {
    int result{0};
    for (auto operation : operations) {
      if (operation == "++X" || operation == "X++") {
        ++result;
      } else {
        --result;
      }
    }

    return result;
  }
};

File: leetcode/2011/final-value-of-variable-after-performing-operations.py
from typing import List, Literal


Operation = Literal["++X", "X++", "--X", "X--"]


class Solution:
    def finalValueAfterOperations(self, operations: List[Operation]) -> int:
        x = 0
        for operation in operations:
            match operation:
                case "++X" | "X++":
                    x += 1
                case "--X" | "X--":
                    x -= 1

        return x
